gg.minion.Minion.yml

```diff
@@ -0,0 +1,78 @@
+      - cp -r {bin,lib,conf} /app/java/jre
+    sources:
+      - type: archive
+        url: https://download.bell-sw.com/java/11.0.17+7/bellsoft-jre11.0.17+7-linux-amd64-full.tar.gz
+        sha256: f3578ddb8fc53053d6c12507c7235e777d6a104d85f8b104fa043c81978d0cb9
+        only-arches:
+          - x86_64
+      - type: archive
+        url: https://download.bell-sw.com/java/11.0.17+7/bellsoft-jre11.0.17+7-linux-aarch64-full.tar.gz
+        sha256: 2a2269e04716ddfb29404a3c5ba8a766b744dbac16c8df9c879adf3ea308d942
+        only-arches:
+          - aarch64
+  - name: minion
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 apply-extra        /app/bin/apply_extra
+      - install -D minion.sh              /app/bin/minion.sh
+      - install -D 48.png                 /app/share/icons/hicolor/48x48/apps/gg.minion.Minion.png
+      - install -D 64.png                 /app/share/icons/hicolor/64x64/apps/gg.minion.Minion.png
+      - install -D 96.png                 /app/share/icons/hicolor/96x96/apps/gg.minion.Minion.png
+      - install -D 128.png                /app/share/icons/hicolor/128x128/apps/gg.minion.Minion.png
+      - install -D 96.png                 /app/share/icons/hicolor/256x256/apps/gg.minion.Minion.png
+      - install -D 128.png                /app/share/icons/hicolor/512x512/apps/gg.minion.Minion.png
+      - install -D $FLATPAK_ID.desktop -t /app/share/applications
+    sources:
+      - type: extra-data
+        filename: Minion.zip
+        sha256: 7ed2a5804462b6dec7d0f4f7b1b96b9279b71ed81939ef03f65f28d9e1c72e1e
+        size: 8046835
+        url: https://cdn.mmoui.com/minion/v3/Minion3.0.11-java.zip
+        # Not using x-checker-data as the used version here is NOT available on the site
+        # but it IS added by the developers exclusively for the flatpak.
+        #
+        # The version for Minion3 will not change as the developers are working on Minion4
+        # and the flatpak will need to be manually updated for Minion4 support when it comes out.
+        #
+        # x-checker-data:
+        #   type: html
+        #   url: https://minion.mmoui.com/?download
+        #   pattern: "href=\"(https://cdn.mmoui.com/minion/v3/(Minion\\d+.\\d+.\\d+)-java.zip)\""
+      - type: script
+        dest-filename: apply-extra
+        commands:
+          - unzip Minion.zip
+          - mv Minion*-java/* .
+          - rm -rf __MACOSX Minion3.0.11-java
+          - rm Minion.zip
+      - type: dir
+        path: icons
+      - type: file
+        path: gg.minion.Minion.desktop
+      - type: script
+        dest-filename: minion.sh
+        commands:
+          # If someone does press udpate it won't create a null folder + it won't hang the application
+          - cd /tmp
+          - java -jar /app/extra/Minion-jfx.jar
+finish-args:
+  - --env=PATH=$PATH:/app/java/jre/bin:/app/bin
+  - --socket=x11
+  - --socket=fallback-x11
+  - --share=network
+  - --device=dri
+  - --filesystem=host
```

Comment:
this is a very broad filesystem permission

Reply:
This program is primarily made to be ran with wine games and the files can change. I can try and default it to allow steam and lutris, then add additional documentation for using flatseal or something.

Reply:
Actually, after editing to not use the host, Minion doesn't want to work properly. It completely ignores folder selection to manage addons in there. 

Also without `host` it won't work well for installations on custom paths (i.e. installing addons for a game which is on another drive or system linking it).

Reply:
Using the `home` filesystem option should actually work in most scenarios as per [this commit](https://github.com/flathub/flathub/pull/4144/commits/4b5c6d36da55a2edcc90b785f2ccabd3932e0e65). I tried manually adding `~/.minion` but it doesn't want to create the folder (it works if it already exists). Usually people also syslink to the Documents folder so they don't need to go through ~10 different subdirectories so I think only home is fine for this app considering its' usage.

Reply:
Using `--persist=.minion` fixed the issue.